lib/src/header_parser/sub_parsers/var_parser.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:ffigen/src/code_generator.dart';
+import 'package:ffigen/src/header_parser/data.dart';
+import 'package:ffigen/src/header_parser/includer.dart';
+import 'package:logging/logging.dart';
+
+import '../clang_bindings/clang_bindings.dart' as clang_types;
+import '../data.dart';
+import '../utils.dart';
+
+final _logger = Logger('ffigen.header_parser.var_parser');
+
+/// Parses a global variable
+Global? parseVarDeclaration(clang_types.CXCursor cursor) {
+  final type = cursor.type().toCodeGenType();
+  if (type.broadType == BroadType.Unimplemented ||
+      type.getBaseType().isIncompleteStruct ||
+      type.getBaseType().unimplementedReason != null) {
+    _logger.fine(
+        '---- Removed Global, reason: unsupported type: ${cursor.completeStringRepr()}');
+    _logger.warning(
+        "Skipped global variable '${cursor.spelling()}', type not supported.");
+    return null;
+  }
+  if (!shouldIncludeGlobalVar(cursor.usr(), cursor.spelling())) {
+    return null;
+  }
+  final g = Global(
+    originalName: cursor.spelling(),
```

Comment:
You also need to add this global to seen globals, If the global is already seen, you can simply return that.
This is needed since the user can include the same header twice, and when that happens, this would generate 2 bindings (say globalVar and globalVar_1).

 You need to define 2 more functions for Globals in lib/src/header_parser/utils.dart file. Similar to this -
```dart

  void addFuncToSeen(String usr, Func func) {
    _functions[usr] = func;
  }

  Func? getSeenFunc(String usr) {
    return _functions[usr];
  }
```